pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -752,6 +752,11 @@ func awsErrorToTerminalError(err error, msg string) error {
 			return prepareAndReturnError()
 		}
 		switch aerr.Code() {
+		case "InstanceLimitExceeded":
```

Comment:
I am not sure if that makes sense, because this error doesn't get fixed by altering the current's machine spec but instead by deleting instances that may or may not be managed by the machine-controller.

IMHO from the POV of the machine-controller this is a transient error, which we probably should make available  as an event but I wouldn't consider it a terminal error

Reply:
the error is an indicator to the end user that we cannot create more machines unless (a manual interaction):

a) some machines will be removed
b) the limit will be increased  (i'm not sure if this is possible)

I'm calling out to @mrIncompetent, as he created the linked issue.

Reply:
Terminal errors are defined as errors which can only be solved by user interaction.
IMO that is the case here.

Reply:
But this error is a condition of your fleet, not of an individual instance. The machine-controller stops acting on machines that have a terminal errors set, right? This does not make sense in this case, as the error can be resolved without changing anything on the machines spec.

Reply:
I don't think we stop processing when a terminal error is set.
We definitly should not.